### DISCUSSION TITLE
Conditional Assignment Operations

### DISCUSSION THREAD

**Discussion by cchannon** [microsoft/power-fx-host-samples#42] [Q&A] [answered]
I'm working on a PowerFX implementation for handling relatively straightforward business logic units. I expect users to need to perform assignment (Set) operations primarily and to perform them conditionally.

I've reviewed the samples closely and I've played around with the RecalcEngine a bit. So far, I am a bit surprised by how challenging this objective seems to be. The RecalcEngine can't handle Set inside a conditional expression (it says Set is an invalid Function) and the samples seem to bear this out, with the REPL sample openly assuming that Set can only be evaluated if it is the opening of a REPL command (The regex match begins with ```@"^\s*Set```).

So, from what I see so far, what I need to do is Regex match every Set operation and substitute them with some Output value like Set1, Set2, etc. Then eval the expression and match my Output back to the Set operation I swapped out, then run that Set operation. Something kinda like the below:

![image](https://user-images.githubusercontent.com/12833126/200210039-ceaed514-d8bb-48ed-8448-16e65febd116.png)

Does that seem right? I feel like there should be something more direct for this, but I can't seem to find it.

**Comment by cchannon** [comment, marked as answer]
Resolved in product core guthub

**Reply by tjohnston-fcg** [reply]
can you share the issue # there?

**Reply by anderson-joyle** [reply]
https://github.com/microsoft/Power-Fx/discussions/815
